### PR TITLE
Physicsengine

### PR DESCRIPTION
**Pull request by RaananW**
The new physics engine architecture!
Is backwards compatible, so there is no need to change existing code.
Some functions are deprecated.
Documentation is coming soon.

### CODE REVIEW

**Comment by jbousquie** [comment]
WWAAAAAOOOOWWwwww

2016-02-22 21:26 GMT+01:00 Raanan Weber <notifications@github.com>:

File: Tools/Gulp/config.json [hunk 1]
```diff
@@ -83,6 +83,8 @@
       "../../src/PostProcess/babylon.postProcess.js",
       "../../src/PostProcess/babylon.postProcessManager.js",
       "../../src/PostProcess/babylon.passPostProcess.js",
+      "../../src/Physics/babylon.physicsJoint.js",
+      "../../src/Physics/babylon.physicsImpostor.js",
```
Comment (deltakosh) [inline Tools/Gulp/config.json]: don't forget to update the file list on the documentation :)

**Comment by dad72** [comment]
Hi Raanan,
Why Mesh.prototype.**getChildren** has been deleted ? another function replaces it? I use this function to reach all children of objects.

**Comment by RaananW** [comment]
it is not deleted, just moved to an upper class. In this commit it is in AbstractMesh, and in a future commit it will be in the Node class, but it should be transparent to you, as the mesh and abstract mesh are both extentions of the Node class.

**Comment by jbousquie** [comment]
maybe is this the answer ?
http://www.html5gamedevs.com/topic/20650-questions-about-meshgetchildren/?do=findComment&comment=117827

**Comment by dad72** [comment]
Ok, I did not have all followed. thanks guys

**Pull request merged**